### PR TITLE
Update Knowledge Management Documentation

### PR DESCRIPTION
**Pull request by saimachi**
# Update Knowledge Management Documentation

## Details on the issue fix or feature implementation

This PR updates the Knowledge Management agent setup guide, reflecting recent schema changes.

## Confirm the following

- [ ]  I started this PR by branching from the head of the default branch
- [ ]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [ ]  I have included inline docs for my changes, where applicable
- [ ]  I have successfully run a local build
- [ ]  I have provided the required update scripts, where applicable
- [ ]  I have updated relevant docs, where applicable

### CODE REVIEW

**Review by kylebunting — CHANGES_REQUESTED** [review]

File: docs/setup-guides/agents/knowledge-management-agent.md [hunk 1]
```diff
@@ -73,8 +95,11 @@ where:
...
 | `description` | The description of the agent, ensure this description details the purpose of the agent. |
-| `indexing_profile_object_id` | The object ID of the indexing profile resource. |
-| `text_embedding_profile_object_id` | The object ID of the text embedding profile resource. |
+| `display_name` | The title of the agent in the Chat UI dropdown menu. This field is optional. |
+| `vectorization.data_source_object_id` | The object ID of the Data Source resource. |
+| `vectorization.indexing_profile_object_id` | The object ID of the indexing profile resource. |
+| `vectorization.text_embedding_profile_object_id` | The object ID of the text embedding profile resource. |
+| `vectorization.text_partitioning_profile_object_id` | The object ID of the text partitioning profile resource. |
 | `prompt_object_id` | The object ID of the prompt resource. |
 | `language_model` | The language model configuration. This sample demonstrates the usage of the Azure OpenAI language model. |
```
[the remark this answers is not in the document]
Comment (saimachi) [reply docs/setup-guides/agents/knowledge-management-agent.md:104]: I added a deprecation notice.

File: docs/setup-guides/agents/knowledge-management-agent.md [hunk 2]
```diff
@@ -73,8 +95,11 @@ where:
 | `name` | The name of the agent. |
 | `object_id` | The object ID of the agent. Remove this element when creating an agent as this is generated by the Management API. |
 | `description` | The description of the agent, ensure this description details the purpose of the agent. |
-| `indexing_profile_object_id` | The object ID of the indexing profile resource. |
-| `text_embedding_profile_object_id` | The object ID of the text embedding profile resource. |
+| `display_name` | The title of the agent in the Chat UI dropdown menu. This field is optional. |
+| `vectorization.data_source_object_id` | The object ID of the Data Source resource. |
```
[the remark this answers is not in the document]
Comment (kylebunting) [reply docs/setup-guides/agents/knowledge-management-agent.md:100]: Here, I would reword this comment slightly. Maybe refer to the agent as a Knowledge Management agent without an inline context versus the "non-Inline Context Knowledge Management" agent.

**Review by kylebunting — CHANGES_REQUESTED** [review]

**Review by kylebunting — APPROVED** [review]

**Review by joelhulen — APPROVED** [review]

File: docs/setup-guides/agents/knowledge-management-agent.md
Comment (ciprianjichici) [inline docs/setup-guides/agents/knowledge-management-agent.md:1]: The agent structure is not correct, the `vectorization` section is missing multiple properties (see `Common/Models/Agent/KnowledgeManagementAgent.cs`.

**Review by ciprianjichici — APPROVED** [review]
